parser.go

```diff
@@ -77,18 +77,21 @@ func (t *ServiceTemplateDefinition) ParseCsar(zipfile string) error {
 			return r, err
 		}
 		r, err = ioutil.ReadAll(rsc)
-		if err != nil {
-			return r, err
-		}
-		return r, nil
+		return r, err
 	}, ParserHooks{ParsedSTD: noop}) // TODO(kenjones): Add hooks as method parameter
 }
 
-func parseImports(impDefs []ImportDefinition, resolver Resolver, hooks ParserHooks) (ServiceTemplateDefinition, error) {
+func parseImports(baseDir string, impDefs []ImportDefinition, resolver Resolver, hooks ParserHooks) (ServiceTemplateDefinition, error) {
 	var std ServiceTemplateDefinition
 
 	for _, im := range impDefs {
-		r, err := resolver(im.File)
+		imFilePath := im.File
+		temp := filepath.Join(baseDir, imFilePath)
```

Comment:
Instead of setting temp here, let's go with the following:

```go
imFilePath := im.File
if baseDir != "" {
    if temp := filepath.Join(baseDir, imFilePath); isLocalPath(temp) {
        imFilePath = temp
    }
}
```